File: C++/SelectionSort.cpp

#include <iostream>
#include <climits>

using namespace std;


void selection_sort (int a[], int n)
{
    
    for (int i=0; i<n-1; i++) 
    {
        
        int min_index = i;  
        
        for (int j=i; j<n; j++)  
        {
            
            if (a[j] < a[min_index] ) 
            {
                
                min_index = j;   
                
            }
            
        }
        
        swap(a[i], a[min_index]);
        
    }
    
}



int main()
{
    int num, key;
    
    cout << "Enter the number of elements in array: ";
    cin >> num;   
    
    int a[1000];
    
    for (int i = 0; i < num; i++)
    {
        
        cin >> a[i];
        
    }
    
    selection_sort(a,num);
    
    for (int i =0; i<num; i++)
    {
        
        cout << a[i] << " , ";
        
    }
    
    
    
    
    
    return 0;
}

File: Java/selectionSort.java
import java.util.Scanner;

public class selectionSort {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("enter the total no=");
        int size = sc.nextInt();
        int arr[] = new int [size];
        for (int i = 0; i < arr.length; i++) {
            arr [i] = sc.nextInt();   
       }
       // int arr[] ={ 7,6,1,8,2};
        for (int i = 0; i < arr.length; i++) {
            int small = i;
            for (int j = i+1; j < arr.length; j++) {
                if (arr[small]>arr[j]) {   
                    small=j;
                } 
            }
             //swap
             int temp = arr[small];
             arr[small]= arr[i];
             arr[i] =temp;
        }
       
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i]+" ");
        }
    }
    
}
